marklogic-data-hub/src/main/java/com/marklogic/hub/impl/ScaffoldingImpl.java

```diff
@@ -34,19 +34,21 @@
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import org.springframework.web.context.annotation.SessionScope;
 
+import javax.annotation.PostConstruct;
 import java.io.*;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.HashMap;
 import java.util.Map;
 
+@SessionScope
 @Component
 public class ScaffoldingImpl implements Scaffolding {
 
-    @Autowired
-    private HubProject project;
+    private HubProject project = null;
```

Comment:
Why even store this as a field? The class already has a dependency on HubConfig. When a HubProject is needed, why not just call hubConfig.getHubProject? Then we don't need a field, and we don't need a PostConstruct method either.